reactor-netty-core/src/main/java/reactor/netty/FutureMono.java

```diff
@@ -167,6 +177,11 @@ public Context get() {
 		@Override
 		@SuppressWarnings("FutureReturnValueIgnored")
 		public void cancel() {
+			// cancel is not thread safe since we assume that removeListener is thread-safe. That said if we have
+			// concurrent addListener and removeListener and if addListener is after removeListener, the other Thread
+			// after execution addListener should see changes happened before removeListener. Thus, it shell see
```

Comment:
same typo as above: `should`, not `shell`

Reply:
done